### PR TITLE
Add event tracking product page spotlight tour

### LINKED ISSUE #33160: Add Event Tracking for Product Page Spotlight Tour
## Acceptance criteria

Create tracks to enable the reporting of:

* % of users that clicks on "Enable interactive walkthrough"
*  % of users that complete the guided mode without dismissing it
*  % of users that clicks on "Skip guided mode"
*  Which steps the user complete by adding information

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Closes #33160.

Adds these tracks for old product walkthrough:
1. `walkthrough_product_view` when starting walkthrough
1. `walkthrough_product_completed` when either clicking dismiss or `Next` button on last step
2. `walkthrough_product_dismissed` when dismissing walkthrough in steps before the last

Adds these tracks for the new product spotlight tour:
1. `walkthrough_product_view` when starting walkthrough
2. `walkthrough_product_enable_button_click` when clicking on "Enable guided mode" button
1. `walkthrough_product_completed` when either clicking dismiss or `Keep editing` button on last step
3. `walkthrough_product_dismissed` when clicking `x` in steps before the last
1. `walkthrough_product_step_completed` when clicking on `Next` each step

### How to test the changes in this Pull Request:

**Testing tracks on old walkthrough**

1. **Without** enabling the `woocommerce_products_tour` experiment
2. Open your browser console and make sure tracking is enabled
4. Enable tracking debugging with `localStorage.setItem( 'debug', 'wc-admin:*' )`
1. Go to WooCommerce > Home > Add my products > Start with a template > Physical product
5. Observe `wcadmin_walkthrough_product_view` track with props `{ spotlight: no, product_template: physical }`
1. Click on `dismiss` button, observe `wcadmin_walkthrough_product_dismissed` with props `{ step_name: 'old-product-name' }`
2. Refresh the browser to get the walkthrough again
3. Click on `Next` until finish
4. Observe `wcadmin_walkthrough_product_completed` track

**Testing tracks on spotlight tour**

1. Install and activate [WCA Test Helper](https://github.com/woocommerce/woocommerce-admin-test-helper/releases/download/v0.7.6/woocommerce-admin-test-helper.zip)
1. Go to Tools > WCA Test Helper > Features
2. Enable `experimental-product-tour`
1. Go to Tools > WCA Test Helper > Experiments
1. Add `woocommerce_products_tour` to treatment
1. Go to WooCommerce > Home > Add my products > Start with a template > Physical product
4. Observe `wcadmin_walkthrough_product_view` track with props `{ spotlight: yes, product_template: physical }`
1. Click on `x` button, observe `wcadmin_walkthrough_product_dismissed` with props `{ step_name: 'product-name' }`
2. Click on "Enable guided mode", observe `wcadmin_walkthrough_product_enable_button_click` track
3. Click on `Next`, observe `wcadmin_walkthrough_product_step_completed` with props `{ step_name: 'product-name', added_info: 'no' }`
4. Type in something to the product description and click on `Next`
5. Observe `wcadmin_walkthrough_product_step_completed` with props `{ step_name: 'product-description', added_info: 'yes' }` 
6. Click on `Next` until finish
7. Observe `wcadmin_walkthrough_product_completed` track

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33400/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Add track for old walkthrough publish button` 12f2e971b866ea7a2d9bb1658575cc5df381e16c
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: plugins/woocommerce-admin/client/wp-admin-scripts/product-tour/index.tsx [hunk 1]
```diff
@@ -239,20 +258,48 @@ const ProductTour = () => {
...
 		const query = qs.parse( window.location.search.slice( 1 ) );
 		if ( query && query.tutorial === 'true' ) {
 			const intervalId = waitUntilElementTopNotChange(
 				tourConfig.steps[ 0 ].referenceElements?.desktop || '',
-				() => setShowTour( true ),
+				() => {
+					setShowTour( true );
+					recordEvent( 'walkthrough_product_view', {
+						spotlight: 'yes',
+						product_template: 'physical',
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/wp-admin-scripts/product-tour/index.tsx:300]: Hardcode product template until we need to support other templates

**Comment by chihsuan** [comment]
@ilyasfoo Nice work!


> Observe wcadmin_walkthrough_product_view track with props { spotlight: no, product_template: physical }

I assume this should be `{ spotlight: yes, product_template: physical }` when testing tracks on spotlight tour?

> `walkthrough_product_completed` when either clicking dismiss or Keep editing button on last step

 What if a user clicks `publish` button on the last step?

**Comment by ilyasfoo** [comment]
Thank you, @chihsuan!

> I assume this should be { spotlight: yes, product_template: physical } when testing tracks on spotlight tour?

Yes you're right, I've corrected the instructions

> What if a user clicks publish button on the last step?

Good point, I'll add a click handler to make sure we catch that and fire completion

**Comment by ilyasfoo** [comment]
I've added the publish listener to spotlight in [1cbfed9](https://github.com/woocommerce/woocommerce/pull/33400/commits/1cbfed92c72dff377a58f872cd961f0aa3e4303d).

I've also added publish listener to old walkthrough in [12f2e97](https://github.com/woocommerce/woocommerce/pull/33400/commits/12f2e971b866ea7a2d9bb1658575cc5df381e16c) although it would record as "dismiss" track if it's not the last step.

**Review by chihsuan — APPROVED** [review]
@ilyasfoo Thanks for the great work! All tested all and looks good. 👍

**Pull request merged**